### PR TITLE
Fix typo in `heuristic_euclidian` helper in AStarGrid2D

### PR DESCRIPTION
**Pull request by sadikkuzu**

### CODE REVIEW

**Comment by sadikkuzu** [comment]
I do not know how to overcome [GHA / Linux / Editor with doubles and GCC sanitizers](https://github.com/godotengine/godot/actions/runs/6210600639/job/16862949982?pr=81778) check.

Could you please help me?

[image](https://github.com/godotengine/godot/pull/81778)

[image](https://github.com/godotengine/godot/actions/runs/6210600639/job/16862949982?pr=81778)

**Comment by AThousandShips** [comment]
It's unrelated, random issue:
* https://github.com/godotengine/godot/issues/78749

**Comment by sadikkuzu** [comment]
Oh, ok, thank you @AThousandShips 🙏🏼

**Comment by akien-mga** [comment]
Could you rebase to remove the merge commit? This PR should only have a single commit. I would also suggest rewording the commit message to be more explicit (like I did with the PR title).

**Comment by sadikkuzu** [comment]
Will go on with #82297
